### DISCUSSION TITLE
Not able to view deeply nested source tree

### DISCUSSION THREAD

**Discussion by mohith46** [appium/appium#19631] [General]
Hi,

**Application - Hybrid (React Native)**

I'm stuck with my automation while trying to inspect the core elements where I feel the elements are deeply nested.
It just stops to the outer padding of the element without letting me inspect the actual element
Below is my capabilities set

{
  "appium:udid": "{Device_UDID}",
  "appium:waitForQuiescence": true,
  "appium:newCommandTimeout": 0,
  "appium:bundleId": "{Device_Bundle_Id}",
  "platformName": "IOS",
  "appium:automationName": "XCuiTest",
  "appium:settings[snapshotMaxDepth]": 62,
  "appium:settings[customSnapshotTimeout]": 0
}

Your help is much needed and will be appreciated.

**Comment by KazuCocoa** [comment]
Would `pageSourceExcludedAttributes` https://appium.github.io/appium-xcuitest-driver/5.13/settings/ or https://appium.github.io/appium-xcuitest-driver/5.12/execute-methods/#mobile-source help if you're saying page source result.

If it was find element thing, it could be XCTest framework by Apple internal potentially (then nothing can help as Appium)

**Reply by mohith46** [reply]
Hi @KazuCocoa ,

Our application is "**react-native**" based, so basically I'm not able to locate the deepest element nested on the screen.
I'm setting the caps of snapshotMaxDepth to a maximum of 62 but still not able to locate the elements.
The source tree itself is not rendering with the deeply nested elements.
Are there any other settings or methods that can help me resolve this issue?

**Reply by KazuCocoa** [reply]
Did you try out `pageSourceExcludedAttributes` and exclude visible as the attached documentation?
This is what we can do as Appium side. According to past conversation in https://github.com/appium/appium/issues/14825 , it looks like the framework itself tends to generate complicated view hierarchy, which could cause similar issues in vanilla XCTest framework itself. Then nothing can be done as Appium, unfortunately.

**Comment by agray** [comment]
It was my understanding from about 12 months ago that a future release of  Appium would overcome this issue?  Does release 7.0.0 fix this?

**Reply by KazuCocoa** [reply]
How deeply nested elements can be found might depend on XCTest framework. For example, Apple sets like below limitation in XCTest itself. The AUT might need to reduce the nested UI elements (generally) to improve the UI performance.

> The 60 depth maximum was added to prevent apps from slowing down system performance when UI queries occur during an XCTest UI test. Unfortunately React Native often creates nested hierarchies that are larger than 60 levels of depth on iOS. That said, you should be able to use webdriverIO for testing, which would not face this 60 depth maximum limit.

https://github.com/appium/appium/issues/14825
